web/packages/shared/components/AwsLaunchButton.tsx

```diff
@@ -111,9 +111,20 @@ function RoleItemList({
   closeMenu: () => void;
   onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
 }) {
+  // If multiple accounts exist, include the AccountID in the item.
+  const allAccountIDs = new Set(awsRoles.map(a => a.accountId));
+  const includeAccountIDDisplay = allAccountIDs.size > 1;
+
   const awsRoleItems = awsRoles.map((item, key) => {
-    const { display, arn, name } = item;
+    const { display, arn, name, accountId } = item;
     const launchUrl = getLaunchUrl(arn);
+    let text = display;
+    if (includeAccountIDDisplay) {
+      text = `${accountId}: ${text}`;
+    }
+    if (display !== name) {
+      text = `${text} (${name})`;
+    }
```

Comment:
Could you add stories for this component to showcase different combinations of these values? See [`TextEditor.story.tsx`](https://github.com/gravitational/teleport/blob/0f57125417769fe6fcae43e0c817ffb0a2d46e58/web/packages/shared/components/TextEditor/TextEditor.story.tsx) for an example of how this could look like. `yarn storybook` and then `http://localhost:9002/?path=/story/shared-texteditor--with-buttons` (or your local IP) to access the storybook.